mods/hide-dotfiles-explorer.wh.cpp

```diff
@@ -0,0 +1,320 @@
+// ==WindhawkMod==
+// @id              hide-dotfiles-explorer
+// @name            Hide Dotfiles (Explorer only)
+// @description     Hide dotfiles and folders starting with . in Windows Explorer and Desktop
+// @version         1.0.1
+// @author          @danalec
+// @github          https://github.com/danalec
+// @include         explorer.exe
+// @compilerOptions -lcomctl32 -lole32 -loleaut32 -lshell32 -lshlwapi
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+This mod hides files and folders starting with a dot (.) in Windows Explorer and Desktop, similar to Unix/Linux systems.
+
+![Screenshot](https://i.imgur.com/3PYbQe7.png)
+
+## How It Works
+This mod **filters files directly from directory listings** at the system level.
+It does **not** modify file attributes (like the "hidden" attribute) and **does not** depend on Windows Explorer's "Show hidden files" setting.
+Files are completely hidden from view regardless of Explorer's hidden file visibility settings.
+
+## Key Benefits
+- **Independent of Explorer settings**: Works regardless of whether "Show hidden files" is enabled or disabled
+- **No file attribute modification**: Files remain unchanged on disk
+- **System-wide filtering**: Affects all Explorer views and file dialogs
+- **Performance optimized**: Uses low-level NT API hooks for minimal overhead
+
+## Configuration
+You can exclude specific files from being hidden using the dotfile whitelist, or specify 
+additional files to always hide regardless of whether they start with a dot.
+
+### Filename Matching
+- **Case-insensitive**: Filenames are matched regardless of case (e.g., `.Gitignore` will match `.gitignore`)
+- **Exact matching**: Full filename must match exactly (no partial matches)
+
+### Settings
+- **Dotfile Whitelist**: List of dotfiles to show (e.g., `.gitignore`, `.env`)
+- **Always Hide**: List of filenames to always hide, even if they don't start with a dot (e.g., `desktop.ini`, `Thumbs.db`)
+
+Examples:
+- To show `.gitignore`: add `.gitignore` to dotfile whitelist
+- To hide `desktop.ini`: add `desktop.ini` to always-hide list
+- `.env.local` will not be shown if only `.env` is whitelisted
+
+## Important Notes
+- **No file attributes are modified** - hidden files retain their original attributes
```

Comment:
Optional suggestion: How about adding an option to choose one of the three modes:
* Never show files (current behavior)
* Show files as hidden (add `FILE_ATTRIBUTE_HIDDEN` attribute, Dot Hide behavior)
* Show files as system (add `FILE_ATTRIBUTE_HIDDEN`, `FILE_ATTRIBUTE_SYSTEM` attributes)

I think that it shouldn't be difficult to add, and I believe some users wouldn't want to have the files removed from the listing completely.

For example, with a folder being listed with the system attribute, it won't be shown in most configurations, but it can still be opened from the address bar or Win+R. With the current behavior, there's no way to access dot folders.

What do you think?